tests/inspectdb/tests.py

```diff
@@ -668,3 +668,55 @@ def test_composite_primary_key(self):
         finally:
             with connection.cursor() as cursor:
                 cursor.execute("DROP TABLE %s" % table_name)
+
+
+class TestInspecDbCrossDatabaseRelationships(TestCase):
+    databases = ["default", "other"]
+
+    @skipUnless(connection.vendor == "mysql", "MySQL specific SQL")
+    def test_unsupported_cross_schema_relationship(self):
+        """Cross-schema relationships should be skipped."""
+        this_table_name = "inspectdb_this_table"
+        other_schema = connections["other"].settings_dict["NAME"]
+        other_table_name = "inspectdb_other_schema_table"
+
+        # Create table in other schema.
+        with connections["other"].cursor() as c:
+            c.execute(
+                f"""
+                CREATE TABLE IF NOT EXISTS `{other_table_name}` (
+                    id int(11) NOT NULL auto_increment,
+                    PRIMARY KEY (id)
+                )
+                """
+            )
+
+        # Create table in default schema that references other_schema.
+        with connection.cursor() as c:
+            c.execute(
+                f"""
+                CREATE TABLE IF NOT EXISTS `{this_table_name}` (
+                    id int(11) NOT NULL auto_increment,
+                    `{other_table_name}_id` int(11) NOT NULL,
+                    PRIMARY KEY (id),
+                    FOREIGN KEY (`{other_table_name}_id`)
+                        REFERENCES `{other_schema}`.`{other_table_name}`(id)
+                )
+                """
+            )
+        try:
+            out = StringIO()
+            call_command(
+                "inspectdb",
+                table_name_filter=this_table_name,
+                stdout=out,
+            )
+            output = out.getvalue()
+            self.assertNotIn(f"{other_table_name} = models.ForeignKey", output)
+            self.assertIn(f"{other_table_name}_id = models.IntegerField", output)
```

Comment:
We can test introspection methods directly i.e. `get_relations()` and `get_constraints()`.

Reply:
5a3074ab372b43170933736712608bbdc0675764